fsspec/caching.py

```diff
@@ -510,33 +576,37 @@ def _fetch(self, start, stop):
         return out + super()._fetch(start, stop)
 
 
-class UpdatableLRU:
+class UpdatableLRU(Generic[P, T]):
     """
     Custom implementation of LRU cache that allows updating keys
 
     Used by BackgroudBlockCache
     """
 
-    CacheInfo = collections.namedtuple(
-        "CacheInfo", ["hits", "misses", "maxsize", "currsize"]
-    )
+    class CacheInfo(NamedTuple):
```

Comment:
It feels like this is a different construct, but it turns out to work the same. I didn't know you could express it this way.